chore: update rust.yml + readme to better document dependencies (#50)
Layr-Labs/rust-kzg-bn254: README.md

[![Crate](https://img.shields.io/crates/v/rust-kzg-bn254-prover.svg)](https://crates.io/crates/rust-kzg-bn254-prover)

Implements KZG commitment and proof generation:
- `KZG`: Main struct for creating commitments and generating proofs
- `SRS`: Structured Reference String handling
- Optimized parallel FFT implementations

### [rust-kzg-bn254-verifier](./verifier)

[![Docs](https://docs.rs/rust-kzg-bn254-verifier/badge.svg)](https://docs.rs/rust-kzg-bn254-verifier/latest/rust_kzg_bn254_verifier/)
[![Crate](https://img.shields.io/crates/v/rust-kzg-bn254-verifier.svg)](https://crates.io/crates/rust-kzg-bn254-verifier)

Provides verification functions:
- Single proof verification
- Batch verification for improved efficiency

## Getting Started

For a complete end-to-end example, see the `test_compute_kzg_proof` function in [prover/tests/kzg_test.rs](./prover/tests/kzg_test.rs).

### Building and Benchmark

```bash
# Build all crates
cargo build

# Run benchmarks
cargo bench
```
### Compatibility
1. The project is compatible with Rust 1.75 or later

### Downstream Dependencies

> 1. Arbitrum Nitro uses rust 1.78

Reviewer: where'd you get this from? I see 1.67: https://github.com/OffchainLabs/nitro/blob/22d99e5c67b62af01fefb29d5be879b0ddc05805/arbitrator/Cargo.toml#L25

Author: Looks like they recently updated as well.

> 2. RiscZero ZKVM uses 1.85

Reviewer: where's this from? I don't see any rust-version in https://github.com/risc0/risc0/blob/main/Cargo.toml#L36-L41

Author: Ah yea, i went off of their rust toolchain version.

Reviewer: Ah then prob just add that as link, and comment that its not actually an MSRV. Should prob create an issue on r0 repo though right? Shouldnt all projects have an MSRV specified?

Author: Agreed.

> 3. SP1 ZKVM uses 1.79

Reviewer: ```suggestion
3. SP1 ZKVM uses [1.79](https://github.com/succinctlabs/sp1/blob/81757da015939d8a851d909e8c3df14bdc3b030d/Cargo.toml#L5)
```

### Notes on testing

> If you encounter issue running tests and it failes with the following error:

Reviewer: ```suggestion
If you encounter issue running tests and it fails with the following error:
```

```
error: package `half v2.5.0` cannot be built because it requires rustc 1.81 or newer, while the currently active rustc version is 1.75.0
```

> then run `cargo update half@2.5.0 --precise 2.4.1` to update this transitive dependency to a lower version.

Reviewer: ```suggestion
then run `cargo update half --precise 2.4.1` to downgrade this transitive dependency to a lower version that works with our MSRV.
```

Reviewer: no need to pin to @2.5.0 because likely 2.5.1 or higher versions will be out soon, making this command outdated.
